cmd/mimir/main.go

```diff
@@ -53,21 +53,27 @@ const (
 var testMode = false
 
 type mainFlags struct {
-	ballastBytes         int  `category:"advanced"`
-	mutexProfileFraction int  `category:"advanced"`
-	blockProfileRate     int  `category:"advanced"`
-	useBufferedLogger    bool `category:"advanced"`
-	printVersion         bool
-	printModules         bool
-	printHelp            bool
-	printHelpAll         bool
+	ballastBytes                  int     `category:"advanced"`
+	mutexProfileFraction          int     `category:"advanced"`
+	blockProfileRate              int     `category:"advanced"`
+	useBufferedLogger             bool    `category:"advanced"`
+	rateLimitedLogsEnabled        bool    `category:"advanced"`
+	rateLimitedLogsPerSecond      float64 `category:"advanced"`
+	rateLimitedLogsPerSecondBurst int     `category:"advanced"`
```

Comment:
I would keep it experimental while we... experiment with it. Mention it in `docs/sources/mimir/configure/about-versioning.md` too.

```suggestion
	rateLimitedLogsEnabled        bool    `category:"experimental"`
	rateLimitedLogsPerSecond      float64 `category:"experimental"`
	rateLimitedLogsPerSecondBurst int     `category:"experimental"`
```